### ISSUE TITLE
Performance of `GetMigrations()` extension method over `DatabaseFacade`

### ISSUE THREAD

**Issue by plachor** [dotnet/efcore#30192]
Hi, during bootstrap phase of our microservices we check if all EF migrations have been already applied and if not we crash container. We observe nondeterministic times of execution of  `GetMigrations()` extension method over `DatabaseFacade`.

Usually first call takes almost one second to execute that method (which is a lot). However sometimes it takes even 11 seconds for same code base and this is more time than actually we wish our container to have during bootstrap. 

On the other hand calling `async GetAppliedMigrationsAsync(ct)` usually takes less than 500ms. And this call requires actual call to Postgres. 

Is it known issue or expected ? And can or do you plan to optimize it in future?

### Include provider and version information

EF Core version: 6.0.7
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL | 6.0.7
Target framework: .NET 6.0
Operating system: Linux

**Comment by ajcvickers** [comment]
This issue is lacking enough information for us to be able to fully understand what is happening. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

**Comment by plachor** [comment]
I will try to create a minimal reproduction in upcoming days.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by plachor** [comment]
Hi I'll add just my observations, 

1) I was not able to provide an easy local reproduction 
2) we found out that we were specifying `MigrationsAssembly` which forced loading of already loaded assembly  I believe, could be optimized on your end as well but we just drooped this 

Finally I come to conclusion that this performance downgrade is related to  Kubernetes auto-scaling  and relaying on spot instances. I believe such new spot node  is heavy on load and for very first few seconds affects started containers performance. But this just a theory for now.
